### ISSUE TITLE
Web apps broken in Firefox 124

### ISSUE THREAD

**Issue by thecimot** [filips123/PWAsForFirefox#475]
### Affected Websites

* all website
* ...

### Environment

# System Details Report
---

## Report details
- **Date generated:**                              2024-03-20 01:05:41

## Hardware Information:
- **Hardware Model:**                              Lenovo IdeaPad 3 14IML05
- **Memory:**                                      12,0 GiB
- **Processor:**                                   Intel® Core™ i3-10110U × 4
- **Graphics:**                                    Intel® UHD Graphics (CML GT2)
- **Disk Capacity:**                               1,3 TB

## Software Information:
- **Firmware Version:**                            DXCN45WW
- **OS Name:**                                     Ubuntu 23.10
- **OS Build:**                                    (null)
- **OS Type:**                                     64-bit
- **GNOME Version:**                               45.2
- **Windowing System:**                            Wayland
- **Kernel Version:**                              Linux 6.5.0-26-generic

* Installation method: Deb

*  Extension version: 2.10.1
*  Native version: 2.10.1
*  Runtime version: 124.0
*  Firefox version: 124.0

![Screenshot from 2024-03-20 00-57-39](https://github.com/filips123/PWAsForFirefox/assets/32328321/9907ef98-e5c4-4dce-a9c1-ef903b626435)

**Comment by filips123** [comment]
Firefox 124 broke something... I will try to fix this soon.

**Comment by hofleverancier** [comment]
FIY I'm on Fedora with firefox-123.0.1-1.fc39.x86_64 and are also having this exact problem since today.

**Comment by hofleverancier** [comment]
Possibly there is something else happening. My Fedora install with Firefox 123 has this issue since today, downgrading Firefox to 119 did not fix the problem. Using non-flatpak'd Firefox.

**Comment by Twijg76** [comment]
I think I have the same issue. It randomly started today.
They worked and without restarting my computer or doing anything special, everyone I opened gave a blank screen.
OpenSUSE Tumbleweed
KDE6 on Wayland
Extension version 2.10.1
the runtime used to be out of date but I updated it while troubleshooting to 2.10.1
Firefox version 123.0.1

**Comment by filips123** [comment]
This should be fixed in https://github.com/filips123/PWAsForFirefox/releases/tag/v2.11.0. Please make sure the native part is updated to that release, and then run "Patch profiles and runtime" from the extension settings.

**Comment by Quasar-Kim** [comment]
**Note for those experiencing the issue even after the update**: I updated the native runtime to 2.11.0 but PWAs were not working. I pressed alt > help > about firefox to see the firefox version, then firefox update itself from 123 to 124, which resolved the issue!

**Comment by filips123** [comment]
Oh, I think the fix breaks web apps on Firefox before 124... I will also try to fix this.

**Comment by filips123** [comment]
I restored compatibility with Firefox before 124 in https://github.com/filips123/PWAsForFirefox/releases/tag/v2.11.1. So, it should now be compatible with all recent Firefox versions.

**Comment by botkero** [comment]
I still have a blank window. I don't know what I'm doing wrong. I also got the banner which advised me to update, but I thought I had done it. 
Any help?

```
Project versions:

    Extension version: 2.11.1
    Native version: 2.10.1
    Runtime version: 124.0
    Firefox version: 124.0
```

**Comment by filips123** [comment]
You also need to update the native version to 2.11.1. Which OS are you on?

**Comment by botkero** [comment]
> You also need to update the native version to 2.11.1. Which OS are you on?

Windows, but I fixed it. It seems that the installer did not recognize my current installation and installed itself somewhere else. I did not notice this immediately, but now everything works.

**Comment by user9713** [comment]
I'm experiencing this issue and I've followed all the steps in this bug report. Any advice on how to fix it?

**Comment by elpollitodiablo** [comment]
Ok so I was having the same issues. What I've tried:

- Verified Firefox Version to be 124
- Verified that the browser extension is 2.11.1
- Verified the native extension is 2.11.1
- Used "Patch profiles and runtime"
- Used "reinstall runtime"
- Reinstalled the native extension completely
- Installed a PWA in a new profile
- Multiple restarts of browser and system

I'm on Windows 11 version 23H2 using chocolatey package manager to manage firefoxpwa

**Comment by elpollitodiablo** [comment]
Ok I fixed it. 

For some reason I had a Firefox PWA extension installed in my AppData directory `C:\Users\***\AppData\Local\Programs\FirefoxPWA` which was used by the Firefox extension. Chocolatey however installed it in `C:\Program Files\FirefoxPWA`

First I checked my PATH and the account PATH was referencing the AppData version and the system PATH was referencing the Program Files version (or the other way around, I forgot). So I updated the PATH variable to use the Program Files version, which did nothing (or maybe I should have done a reboot, but I'm lazy and frustrated). So I renamed the folder PWA in my AppData, which broke the browser extension. It did not recognize the native extension anymore. That made it clear that the browser extension seems hard wired to the installation in AppData.

Long story short: I copied everything from Program Files to the AppData directory of forefoxpwa and, lo and behold, it works.

@filips123 maybe there should be an option in the browser extension to set where the native layer is located.

Hope this helps anyone.

**Comment by AxelTerizaki** [comment]
Hello.

I'm having the same issue. I tried creating new pwas with a new profile but to no avail.

I have Firefox 124.0.1 on Windows and the extension is at version 2.11.1

Is there something I'm missing? What's the native version? How do I check the version or update it?

**Comment by AxelTerizaki** [comment]
Oooh yeah, thank you, that did the trick! I only had one version installed though and no Regedit keys after uninstalling, so I only had to uninstall then reinstall and the extension worked again :)

**Comment by Opisek** [comment]
Just got this same issue with firefox 131.0 and firefoxpwa 2.12.4

**Comment by filips123** [comment]
You need to update the native program to 2.12.5. The extension will say it's outdated, but you can ignore this until the new version is approved.
